### ISSUE TITLE
JSDOM Node not working on CI

### ISSUE THREAD

**Issue by cquiroz** [scala-js/scala-js#2378]
With 0.6.9 I'm trying to run my tests using `jsdom`, as rhino is incompatible with jquery which is in my `jsDependencies`. I setup my js modules to use jsdom and now my run tests fine on my development machine

However, when I try to run them on travis CI, I get errors, in one case:

`java.lang.AssertionError: JS test interface sent bad reply: msg:org.scalatest.events.TestSucceeded`

And in another case I get

```
org.scalajs.jsenv.ComJSEnv$ComClosedException: Node.js isn't connected
```

jsdom in set via travis with the configuration

```
install:
  - rm -rf ~/.nvm && git clone https://github.com/creationix/nvm.git ~/.nvm && (cd ~/.nvm && git checkout `git describe --abbrev=0 --tags`) && source ~/.nvm/nvm.sh && nvm install 5.5.0
  - npm install jsdom
```

As requested on the gitter chat I'm submitting this bug and include pointers to the repo:
https://github.com/cquiroz/ocs3/tree/feature/jsdom-test

And the failing travis log
https://travis-ci.org/cquiroz/ocs3/builds/127114253

**Comment by nicolasstucki** [comment]
It is not the first time that I see this issue, it has also appeared in the `SeleniumJSEnv` in [issue 32](https://github.com/scala-js/scala-js-env-selenium/issues/32). This is probably a bug in the test adapter communication channel itself. In both cases this has only happened on Travis, which makes me believe that there is a race condition somewhere. @gzm0 do you have some insight that could help pinpoint the possible source of this issue?

**Comment by nicolasstucki** [comment]
@cquiroz could you try setting `parallelExecution in Test := false` to see if the problem disappears?

**Comment by cquiroz** [comment]
@nicolasstucki I tried and get the same result disabling `parallelExecution`. Is it possible to show what's passed into node for execution?

**Comment by nicolasstucki** [comment]
It would be a bit tricky. But, you could try to create a new `JSEnv` that extend `JSDOMNodeJSEnv` and then also override the `AbstractDOMNodeRunner` inside of it to have access `getJSFiles` which will give you the files that are being executed on Node.

**Comment by cquiroz** [comment]
I have tried all kind of ways to get this to work but still it doesn't work properly. I have the impression that it is a timing issue especially when you have many files in jsdeps, but I don't have hard data to back that up. On the subprojects where there are no jsdeps it tends to work on CI even though is not 100%.

Could it be that parsing the file takes too long and the connection client/server timesout?

**Comment by markschaake** [comment]
This is happening for me as well in a Gitlab CI context (it was mentioned only TravisCI issues had been reported)

**Comment by tindzk** [comment]
Looking at the code, the issue seems to be related to the timeout set [here](https://github.com/scala-js/scala-js/blob/master/js-envs/src/main/scala/org/scalajs/jsenv/nodejs/AbstractNodeJSEnv.scala#L257). Most likely, it is set too low for Travis CI. Is there an easy way to override the value?

**Comment by Alex-At-Home** [comment]
Not sure if this will work for everyone but note that I was running into this issue and worked around it by moving the test stage to "full" (instead of "fast") when running in TravisCI, eg in `Build.scala` something like:

```scala
  val testStage = sys.props.get("testOpt").map {
    case "full" => FullOptStage
    case "fast" => FastOptStage
  }.getOrElse(FastOptStage)

//...eg:
  val myProject = 
     oldProjectSettings
        .jsSettings(
          scalaJSStage in Global := testStage
        )
```

and then I run `sbt -DtestOpt=full` in my `.travis.yml`

**Comment by lloydmeta** [comment]
Just chiming in to say that I've lately been getting frequent sporadic failures like this in Enumeratum's Travis CI builds (seems to have coincided w/ the new Scalacheck integration FWIW) and what @Alex-At-Home suggested worked like a charm.

**Comment by sjrd** [comment]
The error message in the issue description does not appear in the codebase anymore. It disappeared in bf2c835982198310b2793204e6c6c194e624772d, which was a complete rewrite of the Com mechanism. Therefore I'm going to close this issue as "fixed in 0.6.20". It's possible that the underlying issue still exist, but it any case it would surface with different symptoms, so keeping this issue open is not useful.

**Comment by natgabb** [comment]
Hi @sjrd I'm encountering this same error on a newer version of scala-js (0.6.29)
```
[info] EndTestGroupErrorEvent(<classname>, org.scalajs.jsenv.ComJSEnv$ComClosedException: Node.js isn't connected
```
